src/wallet/wallet.cpp

```diff
@@ -3879,7 +3877,7 @@ std::shared_ptr<CWallet> CWallet::Create(interfaces::Chain& chain, const std::st
         }
     }
 
-    if (fFirstRun)
+    if (walletInstance->m_first_run)
```

Comment:
Question: How does one distinguish whether accessing a wallet field (in this case `walletInstance->m_first_run`) requires `cs_wallet` lock or not? I'm asking because `cs_wallet` is described like this:

```cpp
/*
    * Main wallet lock.
    * This lock protects all the fields added by CWallet.
    */
mutable RecursiveMutex cs_wallet;
```

Can anybody shed light on this for my information?

Reply:
Thanks for the review. Since I need to rebase the branch anyway I'm going to incorporate the suggestion

Reply:
I've dropped `m_first_run` altogether